adoc/extensions/sycl_khr_queue_empty_query.adoc

```diff
@@ -0,0 +1,80 @@
+[[sec:khr-queue-empty-query]]
+= sycl_khr_queue_empty_query
+
+This extension allows developers to query the queue's emptiness, meaning if all
+commands submitted to a queue have been completed.
+
+[[sec:khr-queue-empty-query-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+[[sec:khr-queue-empty-query-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_QUEUE_EMPTY_QUERY# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+
+[[sec:khr-queue-empty-query-funct]]
+== New Queue Function to Query Emptiness
+
+This extension adds the following function to the [code]#sycl::queue# class,
+which provides information about the emptiness of the queue.
+
+'''
+
+.[apidef]#queue::khr_empty#
+[source,role=synopsis,id=api:queue-khr-empty]
+----
+bool khr_empty() const
+----
+
+_Returns:_ [code]#true# if all <<command,commands>> enqueued on this queue have
+completed, [code]#false# otherwise.
+
+{note} Since the implementation executes commands asynchronously, the returned
+value is a snapshot in time.
+{endnote}
+
+'''
+
+[[sec:khr-queue-empty-query-example]]
+== Example
+
+The example below demonstrates the usage of this extension.
+
+[source,,linenums]
+----
+#include <iostream>
+#include <algorithm>
+#include <sycl/sycl.hpp>
+
+int main() {
+  // Pool of queues
+  std::vector<sycl::queue> Qs;
+  size_t n_queues = 12;
+  for (size_t i = 0; i < n_queues; i++)
+    Qs.push_back(sycl::queue(sycl::property_list{sycl::property::queue::in_order{}}));
```

Comment:
```suggestion
    Qs.push_back(sycl::queue{});
```

Maybe just this?  It seems like the in-order property isn't really central to this example.

Reply:
If you use `out of order,` you have concurrency inside each queue, so you should not required to do any kind of load balancing or any kind of queue pools, for that matter.

The example was more to mimic what Cuda people do with their in-order streams.

I can add a comment to explain this.

Reply:
What about:
```
// Pool of in-order queues. 
// One needs to load-balance between queues to expose concurrency.
```
But I have no hard feelings about it; I can remove the `in-order` property if it's people's preference.

Reply:
Maybe you could create each queue on a different device?  Then, the example will load balance across different devices.

Reply:
Good idea!

Reply:
Tried with c3330dc1994e